### DISCUSSION TITLE
`eslint.config.ts` (or type definition)?

### DISCUSSION THREAD

**Discussion by scarf005** [eslint/eslint#16599] [Ideas] [answered]
I like how the new [`eslint.config.js`](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new) works, however, it lacks any proper typing, therefore very hard to maintain. for example, `eslintrc` files have json schemas, which makes it less of a pain.

**Comment by kecrily** [comment, marked as answer]
You can try [eslint-define-config](https://github.com/Shinigami92/eslint-define-config) and follow [Complete rewrite of ESLint discussion](https://github.com/eslint/eslint/discussions/16557#top)

**Reply by scarf005** [reply]
looks like it won't support `eslint.config.js` but only `.eslintrc.js` however.

**Reply by kecrily** [reply]
@scarf005 It added support for flat config [as early as eslint-define-config v1.8.0](https://github.com/Shinigami92/eslint-define-config/releases/tag/1.8.0). Its README.md also mentions usage by `defineFlatConfig` function.

**Reply by scarf005** [reply]
...oh, i've just read first few paragraphs and overlooked that. thank you for clarifying.

**Reply by scarf005** [reply]
made a PR to clarify a bit more:  https://github.com/Shinigami92/eslint-define-config/pull/169
